File: sdk/ios/Sources/FrakSDK/FrakSDKVersion.swift
public enum FrakSDKVersion {
    // Hand-written, and gated as a set with every other version site by `check:native-versions`.
    public static let current: String = "1.0.0-beta.3"

    // `@_spi` rather than plain `internal`: this package is distributed as source through
    // SwiftPM, so a consumer compiles against these declarations and needs an explicit opt-in.

    /// Wire plumbing for `HTTPClient`; not merchant API.
    @_spi(FrakInternal)
    public static let headerName: String = "x-frak-sdk-version"

    /// What `headerName` carries. Platform-prefixed: the version alone is identical on both
    /// SDKs, so a fleet of frozen binaries is otherwise indistinguishable on the wire.
    @_spi(FrakInternal)
    public static let headerValue: String = "ios/\(current)"

    /// Wire plumbing for `FrakSDKUI`'s page URLs; not merchant API.
    @_spi(FrakInternal)
    public static let queryParameterName: String = "sdkVersion"
}

File: sdk/android/frak-sdk/src/main/kotlin/id/frak/sdk/FrakSdkVersion.kt
package id.frak.sdk

/**
 * Version of this SDK build, sent on every request. `@JvmStatic val` rather than `const val`: a
 * `const` is inlined into the merchant's bytecode and would report their compile-time version.
 */
public object FrakSdkVersion {
    /** Keep in step with `frak.sdk.version` in `gradle.properties`; the build checks it. */
    @JvmStatic
    public val CURRENT: String = "1.0.0-beta.2"

    /** Wire plumbing for `HttpClient`; not merchant API. */
    @JvmStatic
    @InternalFrakApi
    public val HEADER_NAME: String = "x-frak-sdk-version"

    /**
     * What [HEADER_NAME] carries. Platform-prefixed: the version alone is identical on both
     * SDKs, so a fleet of frozen binaries is otherwise indistinguishable on the wire.
     */
    @JvmStatic
    @InternalFrakApi
    public val HEADER_VALUE: String = "android/$CURRENT"

    /** Wire plumbing for `:frak-sdk-ui`'s page URLs; not merchant API. */
    @JvmStatic
    @InternalFrakApi
    public val QUERY_PARAMETER_NAME: String = "sdkVersion"
}
